src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -94,17 +108,24 @@ private static Command prepareSet(String[] args) throws DukeException {
         if (hasMinArguments(args, 3)) {
             throw new DukeException("Not enough arguments!");
         }
+        if (hasNoTemplate()) {
+            throw new DukeException("No template set yet!");
+        }
         if (!(isComponent(args[1]) && isNumeric(args[2]))) {
             throw new DukeException("Invalid argument");
         }
 
-        return new Command();
+        double value = Double.parseDouble(args[2]);
+        return new SetCommand(template, args[1], value);
     }
 
     private static Command prepareAdd(String[] args) throws DukeException {
         if (hasMinArguments(args, 4)) {
             throw new DukeException("Not enough arguments!");
         }
+        if (hasNoTemplate()) {
+            throw new DukeException("No template set yet!");
+        }
```

Comment:
Was going to ask about if the current template wasn't set, but saw this. This makes sense!

Reply:
We can consider adding more `Exception` classes and functionality later on.